sopt/src/main/scala/dagr/sopt/cmdline/CommandLineParser.scala

```diff
@@ -147,6 +147,12 @@ class CommandLineParser[SubCommand](val commandLineName: String)
   private type SubCommandClass = Class[_ <: SubCommand]
   private type SubCommandGroupClass = Class[_ <: ClpGroup]
 
+  private var _commandLine: Option[String] = None
+
+  /** The command line with all arguments and values.  Should only be called if either [[parseSubCommand()]] or
+    * [[parseCommandAndSubCommand()]] was successful. */
+  def commandLine: String = _commandLine.get
```

Comment:
I'm curious why you don't just expose the `Option[String]` as `commandLine`?  Seems odd to store an option internally then expose a method that will throw an exception.